src/rustllvm/PassWrapper.cpp

```diff
@@ -97,6 +97,55 @@ LLVMRustAddPass(LLVMPassManagerRef PM, Pass *pass) {
     pm->add(pass);
 }
 
+#define GEN_SUBTARGETS \
+    SUBTARGET(X86) \
+    SUBTARGET(ARM) \
+    SUBTARGET(AArch64) \
+    SUBTARGET(Mips) \
+    SUBTARGET(PPC)
+
+#define SUBTARGET(x) namespace llvm {                \
+    extern const SubtargetFeatureKV x##FeatureKV[];  \
+    extern const SubtargetFeatureKV x##SubTypeKV[];  \
```

Comment:
These symbols are public, but unfortunately they are not exposed through any of the public headers. I will try to ask in the LLVM mailing list about a better way to access the feature names, but this was the best option I could find among the currently available alternatives.